researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider.java

```diff
@@ -167,63 +181,90 @@ public boolean isConsented() {
     // TODO: getConsent rid of Consent methods below on the interface. let ConsentManager handle the
     // implementation details and expose leave giveConsent, getConsent, withdrawConsent, and
     // isConsented
+    @Nullable
     @Override
     public ConsentSignatureBody loadLocalConsent(Context context) {
-
-        return createConsentSignatureBody(authenticationManager.getConsentSync(bridgeConfig.getStudyId()));
+        ConsentSignatureBody consent = createConsentSignatureBody(
+                authenticationManager.getConsentSync(bridgeConfig.getStudyId()));
+        logger.debug("loadLocalConsent called, got: ");
+        return consent;
     }
 
     @Override
-    public void saveConsent(Context context, TaskResult consentResult) {
-        giveConsent(bridgeConfig.getStudyId(), createConsentSignature(consentResult));
+    public void saveConsent(Context context, @NonNull TaskResult consentResult) {
+        logger.warn("saveConsent called -- DEPRECATED");
+        giveConsent(bridgeConfig.getStudyId(), createConsentSignature(consentResult))
```

Comment:
My intuition tells me that whatever bug you were trying to fix in saveLocalConsent() is likely to manifest here as well. Would it be better to have saveConsent() call saveLocalConsent(), and then call the server to upload consent? (Or alternatively, refactor it into a helper method that both saveConsent() and saveLocalConsent() call?)